indexer/src/store.rs

```diff
@@ -0,0 +1,1241 @@
+use crate::types::{
+    CellTransaction, LiveCell, LockHashCellOutput, LockHashIndex, LockHashIndexState,
+    TransactionPoint,
+};
+use bincode::{deserialize, serialize};
+use ckb_core::block::Block;
+use ckb_core::transaction::{CellOutPoint, CellOutput};
+use ckb_core::BlockNumber;
+use ckb_db::{
+    rocksdb::{RocksDB, RocksdbBatch},
+    Col, DBConfig, DbBatch, IterableKeyValueDB, KeyValueDB,
+};
+use ckb_shared::shared::Shared;
+use ckb_store::ChainStore;
+use ckb_traits::chain_provider::ChainProvider;
+use log::{error, info, trace};
+use numext_fixed_hash::H256;
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+
+const BATCH_ATTACH_BLOCK_NUMS: usize = 100;
+const SYNC_INTERVAL: Duration = Duration::from_secs(1);
+const COLUMNS: u32 = 4;
+
+/// +---------------------------------+---------------+--------------------------+
+/// |             Column              |      Key      |          Value           |
+/// +---------------------------------+---------------+--------------------------+
+/// | COLUMN_LOCK_HASH_INDEX_STATE    | H256          | LockHashIndexState       |
+/// | COLUMN_LOCK_HASH_LIVE_CELL      | LockHashIndex | CellOutput               |
+/// | COLUMN_LOCK_HASH_TRANSACTION    | LockHashIndex | Option<TransactionPoint> |
+/// | COLUMN_CELL_OUT_POINT_LOCK_HASH | CellOutPoint  | LockHashCellOutput       |
+/// +---------------------------------+---------------+--------------------------+
+
+const COLUMN_LOCK_HASH_INDEX_STATE: Col = 0;
+const COLUMN_LOCK_HASH_LIVE_CELL: Col = 1;
+const COLUMN_LOCK_HASH_TRANSACTION: Col = 2;
+const COLUMN_CELL_OUT_POINT_LOCK_HASH: Col = 3;
+
+pub trait IndexerStore: Sync + Send {
+    fn get_live_cells(&self, lock_hash: &H256, skip_num: usize, take_num: usize) -> Vec<LiveCell>;
```

Comment:
Need to support confirmation condition to make it work in the real world. http://chainquery.com/bitcoin-api/listunspent

Reply:
There is a `block_number` field in each live cell, api users can use this field and tip number to calculate the numbers of confirmation blocks.